internal/provider/harness_docker_resource.go

```diff
@@ -84,13 +84,21 @@ func (r *HarnessDockerResource) Create(ctx context.Context, req resource.CreateR
 	var data HarnessDockerResourceModel
 	var opts []docker.Option
 
-	log := clog.FromContext(ctx)
-
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
+	encodedInvSeed, err := r.store.Encode(data.Inventory.Seed.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("failed to encode inventory seed", err.Error())
+	}
+
+	ctx, err = provider.InventoryLogger(ctx, encodedInvSeed)
+	if err != nil {
+		resp.Diagnostics.AddError("failed to create logger to file", err.Error())
+	}
```

Comment:
maybe hide the encoding logic in the `InventoryLogger()`?

```suggestion
	ctx, err = provider.InventoryLogger(ctx, data.Inventory.Seed.ValueString())
	if err != nil {
		resp.Diagnostics.AddError("failed to create logger to file", err.Error())
	}
```

Reply:
I'm not too fond of "hiding" it because a function in the `provider` package wouldn't have access to the store instance, so I'd have to send the store and the inventory solely for the purpose of encoding.

What other options we have for this?